### PR TITLE
utils_test.libvirt: add function to hotplug/unhotplug cpu

### PR DESCRIPTION
**Pull request by PandaWei**
the hotplug/unhotplug operation can be supported by
setvcpus and qemu-monitor-command.

### CODE REVIEW

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -2369,3 +2369,33 @@ def update_vm_disk_source(vm_name, disk_source_path, source_type="file"):
     logging.debug("The new VM XML:\n%s", vmxml.xmltreefile)
     vmxml.sync()
     return True
+
+
+def hotplug_domain_vcpu(domain, count, by_virsh=True, hotplug=True):
+    """
+    Hotplug/Unhotplug vcpu for domian
```
Comment (will-Do) [inline virttest/utils_test/libvirt.py]: Should be "Hot-plug or hot-unplug"

File: virttest/utils_test/libvirt.py [hunk 2]
```diff
@@ -2369,3 +2369,33 @@ def update_vm_disk_source(vm_name, disk_source_path, source_type="file"):
...
+                     otherwise, using qemu_monitor
+    :param hotplug:  True means hotplug, False means unhotplug
+    """
+    if by_virsh:
+        result = virsh.setvcpus(domain, count, "--live", debug=True)
+    else:
+        if hotplug:
+            cpu_opt = "cpu-add"
+        else:
+            cpu_opt = "cpu-del"
```
Comment (will-Do) [inline virttest/utils_test/libvirt.py]: #vim tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py +247:
```
    ┊   ┊   # setvcpu/hotplug is only available as of qemu 1.5 and it's still
    ┊   ┊   # evolving. In general the addition of vcpu's may use the QMP
    ┊   ┊   # "cpu_set" (qemu 1.5) or "cpu-add" (qemu 1.6 and later) commands.
    ┊   ┊   # The removal of vcpu's may work in qemu 1.5 due to how cpu_set
    ┊   ┊   # can set vcpus online or offline; however, there doesn't appear
    ┊   ┊   # to be a complementary cpu-del feature yet, so we can add, but
    ┊   ┊   # not delete in 1.6.

```

Comment (PandaWei) [inline virttest/utils_test/libvirt.py]: Yes, you are right.
[root@localhost virt-test]# virsh qemu-monitor-command 15 --pretty '{"execute":"cpu-del","arguments":{"id":7}}'
{
    "id": "libvirt-23",
    "error": {
        "class": "CommandNotFound",
        "desc": "The command cpu-del has not been found"
    }
}

File: virttest/utils_test/libvirt.py (hunk 1 above)
Comment (PandaWei) [inline virttest/utils_test/libvirt.py]: Thanks for your reminder.

**Comment by PandaWei** [comment]
Updated it.

File: virttest/utils_test/libvirt.py (hunk 2 above)
Comment (PandaWei) [inline virttest/utils_test/libvirt.py]: I think it's better that check the result at test case.
so i add some explanation.

**Comment by will-Do** [comment]
OK

**Comment by liyangfnst** [comment]
ACK, merge it now.

**Pull request merged**